src/translator/service.h

```diff
@@ -1,146 +1,116 @@
+  /// information like quality-scores, alignments etc.
 
-  /// Returns if model is alignment capable or not.
-  bool isAlignmentSupported() const { return options_->hasAndNotEmpty("alignment"); }
+  /// If you have async/multithread capabilities, it is recommended to work with AsyncService instead of this class.
+  /// Note that due to batching differences and consequent floating-point rounding differences, this is not guaranteed
+  /// to have the same output as AsyncService.
+
+  /// @param [in] translationModel: TranslationModel to use for the request.
+  /// @param [in] source: rvalue reference of the string to be translated
+  /// @param [in] responseOptions: ResponseOptions indicating whether or not to include some member in the Response,
+  /// also specify any additional configurable parameters.
+  std::vector<Response> translateMultiple(std::shared_ptr<TranslationModel> translationModel,
+                                          std::vector<std::string> &&source, const ResponseOptions &responseOptions);
 
  private:
-  /// Queue an input for translation.
-  void queueRequest(std::string &&input, std::function<void(Response &&)> &&callback, ResponseOptions responseOptions);
+  ///  Numbering requests processed through this instance. Used to keep account of arrival times of the request. This
+  ///  allows for using this quantity in priority based ordering.
+  size_t requestId_;
 
-  /// Translates through direct interaction between batcher_ and translators_
+  /// An aggregate batching pool associated with an async translating instance, which maintains an aggregate queue of
+  /// requests compiled from  batching-pools of multiple translation models. Not thread-safe.
+  AggregateBatchingPool batchingPool_;
 
-  /// Number of workers to launch.
-  size_t numWorkers_;
+  Config config_;
+};
 
-  /// Options object holding the options Service was instantiated with.
-  Ptr<Options> options_;
+/// Effectively a threadpool, providing an API to take a translation request of a source-text, paramaterized by
+/// TranslationModel to be used for translation. Configurability on optional items for the Response corresponding to a
+/// request is provisioned through ResponseOptions.
+class AsyncService {
+ public:
+  struct Config {
+    size_t numWorkers;
+  };
+  /// Construct an AsyncService with configuration loaded from Options. Expects positive integer value for
+  /// `cpu-threads`. Additionally requires options which configure AggregateBatchingPool.
+  AsyncService(const AsyncService::Config &config);
+
+  /// Create a TranslationModel compatible with this instance of Service. Internally assigns how many replicas of
+  /// backend needed based on worker threads set. See TranslationModel for documentation on other params.
+  template <class ConfigType>
+  Ptr<TranslationModel> createCompatibleModel(const ConfigType &config, MemoryBundle &&memory = MemoryBundle{}) {
```

Comment:
Since your `TranslationModel` already has elements of service embedded (the batching pool), just make `TranslationModel` take the `AsyncService` or `BlockingService` in the constructor.

Reply:
Consensus in meeting if I understood correctly is to solve this via separating workspaces and binding notions of number of workers with Service. 

Alternatives discussed where overloading with both Service and packing it behind TranslationModel, but this would make pivoting rather unintuitive.